search-sentiments-map/test/search-test.js

```diff
@@ -38,37 +223,218 @@ describe('Search', function() {
 
     it('should update search results for all global trends', async function() {
       await search.updateSearchResults();
-      deleteAncientResultsStub.callCount.should.equal(1);
-      retrieveGlobalTrendsStub.callCount.should.equal(1);
-      getSearchResultsForCountriesForTopicStub.getCall(0).args[1].should.equal('trend1');
-      getSearchResultsForCountriesForTopicStub.getCall(1).args[1].should.equal('trend2');
-      getSearchResultsForCountriesForTopicStub.callCount.should.equal(2);
-      sleepStub.callCount.should.equal(2);
+      assert.equal(deleteAncientResultsStub.callCount, 1);
+      assert.equal(sleepStub.callCount, globalTrends.length);
+      for (let i = 0; i < datastoreEntities.length; i++) {
+        assert.equal(datastoreEntities[i].key, 'fakeKey');
+        assert.equal(datastoreEntities[i].data.topic,
+            globalTrends[i].trendTopic);
+        assert.equal(datastoreEntities[i].data.lowercaseTopic,
+            globalTrends[i].trendTopic.toLowerCase());
+      }
     });
   });
 
-  describe('RetrieveSearchResultFromDatastore', function() {
+  describe('GetSearchResultsForCountriesForTopic', function() {
+    let mockSearchResultData;
+
     beforeEach(() => {
-      sinon.stub(Datastore.prototype, 'runQuery').callsFake(() => {
-        results = [{
-          topic: 'testTopic',
-          timestamp: 0,
-          dataByCountry: {},
-        }];
-        return [results];
+      // Function is stubbed because it is tested separately.
+      sinon.stub(search, 'formatCountryResults').callsFake((searchResults) => {
+        mockSearchResultData ={
+          title: 'mockTitle',
+          snippet: 'mockSnippet',
+          htmlTitle: 'mockHtmlTitle',
+          link: 'mockLink',
+          score: 100,
+        };
+
+        let results = []
+        searchResults.items.forEach(result => {
+          results.push(mockSearchResultData);
+        });
+        return Promise.resolve({
+          score: 100,
+          results: results,
+        });
       });
+              
+      // Function is stubbed because it is tested separately.
+      sinon.stub(searchInterestsModule, 'getGlobalSearchInterests').resolves([
+        {
+          geoCode: 'AU',
+          value: [10],
+        },
+        {
+          geoCode: 'US',
+          value: [10],
+        }
+      ]);
+
+      // Sleep function is stubbed to avoid 1 minute pauses.
+      sleepStub = 
+          sinon.stub(search, 'sleep').resolves('Not interested in the output');
+
+      // Stub Custom Search API call.
+      let json = {
+        items: [
+          {
+            title: 'mockTitle',
+            snippet: 'mockSnippet',
+            htmlTitle: 'mockHtmlTitle',
+            link: 'mockLink'
+          }
+        ]
+      };
+
+      let responseObject = {
+        status: "200",
+        json: () => {
+          return json;
+        }
+      };
```

Comment:
Id recommend a constant "success response" or something. You currently have just one use right now, but it would come in handy if you add more tests.

Reply:
So the problem is the json it returns would be different each time depending on the function that it is returning from.